src/app/common/loading-modal/loading-dialog.service.ts

```diff
@@ -0,0 +1,22 @@
+import {Injectable} from '@angular/core';
+import {MatSnackBar, MatSnackBarRef} from '@angular/material/snack-bar';
+import {LoadingModalComponent} from './loading-modal.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoadingDialogService {
+
+  public constructor(private snackBar: MatSnackBar) {
+  }
+
+  private snackBarRef!: MatSnackBarRef<LoadingModalComponent>;
```

Comment:
Here probably you should not use `!`. Of course we can assume everyone will use it correctly but still can happen that for example someone will call somehow `stopLoading` without calling first `startLoading`. In that case this should be:
```typescript
private snackBarRef: MatSnackBarRef<LoadingModalComponent> | undefined;
```